include/dfu/flash_img.h

```diff
@@ -36,13 +36,26 @@ struct flash_img_check {
 /**
  * @brief Initialize context needed for writing the image to the flash.
  *
+ * This API call is deprecated and flash_img_init_area should be used instead.
+ *
  * @param ctx     context to be initialized
  * @param area_id flash area id of partition where the image should be written
  *
  * @return  0 on success, negative errno code on fail
  */
 int flash_img_init_id(struct flash_img_context *ctx, uint8_t area_id);
```

Comment:
need to design API using  `__deprecated`
This remark is also valid to any else deprecated API function.

Reply:
Actually I have left the comment but I am not sure about deprecation. Both functions can coexist, of course I would like people to use the `flash_img_init_area` rather than the `flash_img_init_id` but it does not seem to hurt to have both. Should we deprecate it @nvlsianpu ?

Reply:
You sand that the call is deprecated above - that why I requested the change.
Seams deprecation is not really your intention, let's change the comment to reflect that:
`This API call is kept for backward compatibility. flash_img_init_area() is recommended to be used instead.`